### ISSUE TITLE
[feature request] Multidim support for softmax/log_softmax/softmin 

### ISSUE THREAD

**Issue by vadimkantorov** [pytorch/pytorch#71357]
### 🚀 The feature, motivation and pitch

This was lost in comments in https://github.com/pytorch/pytorch/issues/9703#issuecomment-407137379 and https://github.com/pytorch/pytorch/issues/12316#issuecomment-433362779, and still seems not implemented

Real-world usecase that could be simplified: https://github.com/fundamentalvision/Deformable-DETR/blob/main/models/ops/modules/ms_deform_attn.py#L99-L100
